src/lang/proc_macros/controller.rs

```diff
@@ -208,14 +184,25 @@ impl ProcMacroClientController {
     /// except the one in the [`ProcMacroInput`].
     #[tracing::instrument(level = "trace", skip_all)]
     pub fn force_restart(&mut self, db: &mut AnalysisDatabase, config: &Config) {
-        self.remove_all_proc_macros_from_db(db);
-        self.crate_plugin_suites.clear();
+        for (crate_input, plugins) in mem::take(&mut self.crate_plugin_suites) {
+            db.remove_crate_plugin_suite(crate_input, plugins);
+        }
         db.reset_proc_macro_resolutions();
         self.clean_up_previous_proc_macro_server(db);
-
         self.try_launch_proc_macro_server(db, config);
     }
 
+    /// Requests the proc-macro-server to load the workspace with the given `manifest_path`
+    /// and respond with all macros available for it.
```

Comment:
```suggestion
    /// Requests the proc-macro-server to: load the workspace with the given `manifest_path`
    /// and respond with all macros available for it.
```
Otherwise it sounds like we (client) should respond.